### PR TITLE
Local server: split apart the constructor

### PR DESCRIPTION
**Pull request by sdemjanenko**
This method was getting pretty hard to follow.

### CODE REVIEW

File: lib/local-server/index.js [hunk 1]
```diff
@@ -16,46 +16,73 @@ module.exports = class Server {
...
+          pageType = 'head'
+        }
+      }
+
+      if (pageType) {
+        resp.end(Zen.indexHtml(pageType))
+      } else {
+        next()
+      }
+    })
```
Comment (rogueg) [inline lib/local-server/index.js:58]: This seems much bigger than the 1 line it was previously.

Comment (sdemjanenko) [reply lib/local-server/index.js:58]: As far as I can tell the previous 1 line was sending `head.js` to any request that failed to match a route. This new code should send a 404 if an unknown asset is requested.
